src/main/kotlin/org/rust/lang/core/psi/ext/RsMacroCall.kt

```diff
@@ -105,7 +111,8 @@ fun RsMacroCall.findIncludingFile(): RsFile? {
     val path = includeMacroArgument?.expr?.value ?: return null
     // TODO: it doesn't work if `include!()` macro call comes from other macro
     val file = containingFile?.originalFile?.virtualFile ?: return null
-    return file.parent?.findFileByRelativePath(path)?.toPsiFile(project)?.rustFile
+    // It's important to refresh before find because file can be created externally (for example, while compilation)
+    return file.parent?.refreshAndFindFileByMaybeRelativePath(path)?.toPsiFile(project)?.rustFile
```

Comment:
`refreshAndFindFileByPath` can be invoked outside of the read action or from EDT, otherwise, it can lead to deadlock. But looks like this code can be invoked from background thread in the read action

Reply:
Nice catch!
Thanks